src/bundle.c

```diff
@@ -601,26 +608,47 @@ enum swupd_code remove_bundles(char **bundles)
 		}
 
 		/* check if bundle is required by another installed bundle */
-		struct list *reqd_by = NULL;
-		required_by(&reqd_by, bundle, current_mom, 0);
-		if (reqd_by != NULL) {
+		struct list *tree_reqd_by = NULL;
+		struct list *simple_reqd_by = NULL;
+
+		required_by(&simple_reqd_by, bundle, current_mom, 0, TREE_OFF);
+		if (simple_reqd_by != NULL) {
+			list_str_deduplicate(simple_reqd_by);
+			int number_of_reqd = list_len(simple_reqd_by);
+			/* let's try to build a dependency tree */
+			required_by(&tree_reqd_by, bundle, current_mom, 0, TREE_ON);
```

Comment:
We need now to execute required_by() function twice for every bundle that we want to install just to decide if we will print the full list or not. Not sure if it's something that we want.
What about only printing the list of dependent bundles and printing the full list only when --verbose is used?

Reply:
that is something we can do..... now that we have the --verbose flag, instead of sowing one way or the other based on number of tree branches we can just always default for the simple way and only show the tree view if the user uses --verbose. I like that idea